packages/react-theme-provider/src/context.ts

```diff
@@ -0,0 +1,11 @@
+import { Theme } from '@fluentui/react-theme';
+import * as React from 'react';
+
+export interface ThemeProviderValue extends Theme {}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const internal__ThemeContext = React.createContext<ThemeProviderValue>(({} as unknown) as Theme);
```

Comment:
Why is this `internal__` ? Users who are still on class components would need to use this instead of `useTheme`.
Also, shouldn't default value be default (web light) theme here?

Reply:
I think the reason is to not bundle the theme with the provider

Reply:
https://github.com/microsoft/fluentui/pull/16762#discussion_r568790161

---

Overall, I don't think that we should allow using contexts directly as it also leaks `.Consumer` and `.Provider` members that customers can consume. It's a valid point about class components, but at the same time - `makeStyles()` currently has only hooks version...

---

We can use:

```
delete ThemeContext.Consumer
delete ThemeContext.Provider
```

But it's still opens a way to consume via `React.useContext()` while `useTheme()` hook is preferred as it's a part of public API. Any thoughts?

Reply:
I don't have better idea. Let's leave this as is for now